lib/linklist.c

```diff
@@ -150,6 +150,21 @@ bool listnode_add_sort_nodup(struct list *list, void *val)
 	return true;
 }
 
+struct list *list_dup(struct list *list)
+{
+	struct list *dup;
+	struct listnode *node;
+	void *data;
+
+	assert(list);
+
+	dup = list_new();
```

Comment:
Should we copy the cmp and del functions?

Reply:
Good catch. Done.